pkg/kubelet/lifecycle/predicate.go

```diff
@@ -247,5 +250,16 @@ func GeneralPredicates(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) ([]Pr
 		reasons = append(reasons, &PredicateFailureError{nodeports.Name, nodeports.ErrReason})
 	}
 
+	// Check taint/toleration except for static pods
+	if !types.IsStaticPod(pod) {
+		_, isUntolerated := corev1.FindMatchingUntoleratedTaint(nodeInfo.Node().Spec.Taints, pod.Spec.Tolerations, func(t *v1.Taint) bool {
+			// PodToleratesNodeTaints is only interested in NoSchedule and NoExecute taints.
+			return t.Effect == v1.TaintEffectNoSchedule || t.Effect == v1.TaintEffectNoExecute
+		})
+		if isUntolerated {
```

Comment:
how does TolerationSeconds interact with this evaluation?

Reply:
Any toleration (even with tolerationSeconds) applies to a taint at scheduling/startup time.

The `tolerationSeconds` is about delaying the eviction.

Reply:
is there a reason we cant move this check inside `scheduler.AdmissionCheck` after it validates nodeports for example?